python/cudf/cudf/dataframe/multiindex.py

```diff
@@ -555,18 +556,23 @@ def to_pandas(self):
         pandas_codes = []
         for code in self.codes.columns:
             pandas_codes.append(self.codes[code].to_array())
+        pandas_levels = [
+            level.to_pandas() if hasattr(level, "to_pandas") else level
+            for level in self.levels
+        ]
         # Backwards compatibility:
         # Construct a dummy MultiIndex and check for the codes attr.
         # This indicates that it is pandas >= 0.24
         # If no codes attr is present it is pandas <= 0.23
         if hasattr(pd.MultiIndex([[]], [[]]), "codes"):
-            return pd.MultiIndex(
-                levels=self.levels, codes=pandas_codes, names=self.names
-            )
+            pandas_mi = pd.MultiIndex(levels=pandas_levels, codes=pandas_codes)
         else:
-            return pd.MultiIndex(
-                levels=self.levels, labels=pandas_codes, names=self.names
+            pandas_mi = pd.MultiIndex(
+                levels=pandas_levels, labels=pandas_codes
             )
```

Comment:
Pandas dependency is bumped to >=0.24.2, <0.25 so you should be fine to clean this up.

Reply:
I'd like to do that as part of another PR if you don't mind. This change is actually in support of the _repr_ PR, casting between CUDF and Pandas MultiIndexes more correctly.